container/go/pkg/oci/write_test.go

```diff
@@ -0,0 +1,188 @@
+package oci
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+)
+
+var (
+	blob     = "blobs/"
+	index    = "index.json"
+	ocif     = "oci-layout"
+	testPath = "testdata/test_write1"
+)
+
+var testCases = []struct {
+	// name is the name of the test case
+	name string
+	// dataPath is where the test data is found
+	dataPath string
+	// the sha code for the manifest
+	shaPath string
+	// content is the oci layout content where the key is the file path and value is sha256 sum
+	content map[string]v1.Hash
+}{
+	{
+		"alpine_linux",
+		"testdata/test_write1/small_linux.tar",
+		"sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c",
+		map[string]v1.Hash{
+			"blobs/sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c": v1.Hash{Algorithm: "sha256", Hex: "9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c"},
+			"index.json": v1.Hash{Algorithm: "sha256", Hex: "ce6ed2e6817add8a5a0e94f4c2ac00360c7ca2cc98c4972b86530f14de622508"},
+			"oci-layout": v1.Hash{Algorithm: "sha256", Hex: "b66dbb27a73334db6ac9c030475837bd7f4472d835c72b2360534b203edce6cb"},
+		},
+	},
+}
+
+// TestWrite checks to ensure that the v1.Image written is in OCI format by validating image output format and content
+func TestWrite(t *testing.T) {
+	for _, wt := range testCases {
+		t.Run(wt.name, func(t *testing.T) {
+			tmp, err := ioutil.TempDir("", "write-oci-index-test")
+			if err != nil {
+				t.Errorf("Write(%s): %v", os.TempDir(), err)
+			}
+
+			defer os.RemoveAll(tmp)
+
+			img, err := tarball.ImageFromPath(wt.dataPath, nil)
+			if err != nil {
+				t.Errorf("error loading image from testdata: %v", err)
+			}
+
+			if err := assertImageIndexValid(tmp, img); err != nil {
+				t.Errorf("image written is not a valid index: %v", err)
+			}
+
+			if err = assertOCIFormat(tmp, img, wt.content, wt.shaPath); err != nil {
+				t.Errorf("image is not in oci format: %v", err)
+			}
+		})
+	}
+}
+
+func assertImageIndexValid(tmp string, img v1.Image) error {
+	if err := Write(img, tmp); err != nil {
```

Comment:
It's a little weird having this write action in this function. I would probably put this Write() call in TestWrite above directly. And then have one single function called in the TestWrite to do all the checks, maybe called assertValidImageOCILayout() which contains the rest of assertImageIndexValid + assertOCIFormat. So basically, rename assertOCIFormat and move the rest of assertImageIndexValid there.